Sources/MapboxMaps/Location/InterpolatedLocationProducer.swift

```diff
@@ -40,11 +52,47 @@ internal final class InterpolatedLocationProducer: NSObject, InterpolatedLocatio
         }
     }
 
+    // MARK: Puck Location Consumers.
+
     // delivers the latest location synchronously, if available
     internal func observe(with handler: @escaping (InterpolatedLocation) -> Bool) -> Cancelable {
         return observableInterpolatedLocation.observe(with: handler)
     }
 
+    private var hasPuckLocationConsumers: Bool {
+        consumers.count > 0
+    }
+
+    internal func addPuckLocationConsumer(_ consumer: PuckLocationConsumer) {
+        consumers.add(consumer)
+        syncConsumers()
+    }
+
+    internal func removePuckLocationConsumer(_ consumer: PuckLocationConsumer) {
+        consumers.remove(consumer)
+        syncConsumers()
+    }
+
+    private func syncConsumers() {
+        guard isEnabled, hasPuckLocationConsumers else {
+            cancelableToken?.cancel()
+            cancelableToken = nil
+            return
+        }
+
+        if cancelableToken == nil {
+            cancelableToken = observableInterpolatedLocation.observe(with: { [weak self] interpolatedLocation in
+                guard let self = self else { return false }
+                for puckLocationConsumer in self.consumers.allObjects {
```

Comment:
nit: It might be just me, but I prefer the new line just after the `guard let self …`.
WDYT? 
/cc @evil159 @maios 
```suggestion
                guard let self = self else { return false }
                
                for puckLocationConsumer in self.consumers.allObjects {
```